obsidian.el

```diff
@@ -451,7 +451,11 @@ Argument S relative file name to clean and convert to absolute."
   "Filter ALL-FILES to return list with same name as F."
   (-filter (lambda (el) (s-ends-with-p f el)) all-files))
 
-(defun obsidian-find-file (f)
+(defun obsidian--find-file-with-window (file-name &optional arg)

```

Comment:
You're on a roll :) in this case you didn't need to extract into a separate function, because all you want changed depending on the `arg`, so it can be simplified to just a change of symbol:

```elisp
(let* ((find-fn (if arg #'find-file-other-window #'find-file)))
  (funcall find-fn "~/Downloads/transactions-blaster-to-jun092023.csv"))
```

Reply:
Ha yeah I've been trying to get my obsidian workflow settled so just trying to iron out the remaining kinks; luckily you've already done all the heavy lifting.

Your solution is definitely cleaner so I'll make that change, thanks for the feedback.

Reply:
I wanted to keep your threading syntax but kept getting a void-function error so there's clearly something I'm not understanding with that syntax.  I ended up using the `funcall` syntax that you suggested.

``` emacs-lisp
(-> file obsidian--expand-file-name find-fn)
=> Debugger entered--Lisp error: (void-function find-fn)
```

Reply:
This probably has something to do with Elisp being lisp-2. I think you can thread last like this:

`(->> file (funcall find-fn))`